lib/internal/warning.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+// A process warning is esssentially just an Error object emitted through the
+// process.on('warning') event rather than being thrown.
+
+const prefix = `(${process.release.name}:${process.pid}) `;
+
+function inherits(ctor, superCtor) {
+  ctor.super_ = superCtor;
+  Object.setPrototypeOf(ctor.prototype, superCtor.prototype);
+}
+
+function Warning(name, message, constructor) {
```

Comment:
it may not be strictly necessary right now but I would think it's safest to insert `Error.call(this)` in this constructor